core/filesystem/directories.hpp

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_FILESYSTEM_DIRECTORIES_HPP
+#define KAGOME_FILESYSTEM_DIRECTORIES_HPP
+
+#include "filesystem/common.hpp"
+
+namespace kagome::filesystem {
+
+  /**
+   * Creates directories recursively
+   * @param path full path or relative path to directory to be created
+   * @return true if path exists, otherwise false
+   */
+  inline bool createDirectoryRecursive(const path &path) {
+    return exists(path) || create_directories(path);
```

Comment:
This overload may throw, I suggest using an overload that returns an error code.

Reply:
Yep! We wanted anyway to catch all the exceptions because of multiple throws. In this case, we can display a description of the error without repeating the codes in our code.